commands/transfer_funds.go

```diff
@@ -144,3 +123,129 @@ func checkTransfer(cmd *commandspb.Transfer) (e Errors) {
+		errs.AddForProperty(prefix+".individual_scope", ErrIsNotValid)
+	}
+	if dispatchStrategy.DistributionStrategy == vega.DistributionStrategy_DISTRIBUTION_STRATEGY_UNSPECIFIED && toAccountType != vega.AccountType_ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS {
+		errs.AddForProperty(prefix+".distribution_strategy", ErrIsRequired)
+	}
+	if dispatchStrategy.DistributionStrategy != vega.DistributionStrategy_DISTRIBUTION_STRATEGY_UNSPECIFIED && toAccountType == vega.AccountType_ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS {
+		errs.AddForProperty(prefix+".distribution_strategy", ErrIsNotValid)
+	}
+	if len(dispatchStrategy.StakingRequirement) > 0 {
+		if toAccountType == vega.AccountType_ACCOUNT_TYPE_REWARD_VALIDATOR_RANKING || toAccountType == vega.AccountType_ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS {
+			errs.AddForProperty(prefix+".staking_requirement", ErrIsNotValid)
+		} else if staking, ok := big.NewInt(0).SetString(dispatchStrategy.StakingRequirement, 10); !ok {
+			errs.AddForProperty(prefix+".staking_requirement", ErrNotAValidInteger)
+		} else if staking.Cmp(big.NewInt(0)) < 0 {
+			errs.AddForProperty(prefix+".staking_requirement", ErrMustBePositiveOrZero)
+		}
+	}
+	if len(dispatchStrategy.NotionalTimeWeightedAveragePositionRequirement) > 0 {
+		if toAccountType == vega.AccountType_ACCOUNT_TYPE_REWARD_VALIDATOR_RANKING || toAccountType == vega.AccountType_ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS {
+			errs.AddForProperty(prefix+".notional_time_weighted_average_position_requirement", ErrIsNotValid)
+		} else if notional, ok := big.NewInt(0).SetString(dispatchStrategy.NotionalTimeWeightedAveragePositionRequirement, 10); !ok {
+			errs.AddForProperty(prefix+".notional_time_weighted_average_position_requirement", ErrNotAValidInteger)
+		} else if notional.Cmp(big.NewInt(0)) < 0 {
+			errs.AddForProperty(prefix+".notional_time_weighted_average_position_requirement", ErrMustBePositiveOrZero)
+		}
+	}
+	if dispatchStrategy.WindowLength > 0 && toAccountType == vega.AccountType_ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS {
+		errs.AddForProperty(prefix+".window_length", ErrIsNotValid)
+	}
+	if dispatchStrategy.WindowLength > 100 {
+		errs.AddForProperty(prefix+".window_length", ErrMustBeAtMost100)
+	}
+	if len(dispatchStrategy.RankTable) == 0 && dispatchStrategy.DistributionStrategy == vega.DistributionStrategy_DISTRIBUTION_STRATEGY_RANK {
+		errs.AddForProperty(prefix+".rank_table", ErrIsNotValid)
+	}
+	if len(dispatchStrategy.RankTable) > 0 && dispatchStrategy.DistributionStrategy != vega.DistributionStrategy_DISTRIBUTION_STRATEGY_RANK {
+		errs.AddForProperty(prefix+".rank_table", ErrIsNotValid)
+	}
+	if len(dispatchStrategy.RankTable) > 500 {
+		errs.AddForProperty(prefix+".rank_table", ErrIsNotValid)
+	}
+	if len(dispatchStrategy.RankTable) > 1 {
+		for i := 1; i < len(dispatchStrategy.RankTable); i++ {
+			if dispatchStrategy.RankTable[i].StartRank <= dispatchStrategy.RankTable[i-1].StartRank {
+				errs.AddForProperty(prefix+".rank_table", ErrIsNotValid)
```

Comment:
we need to specify which item is the problem, otherwise it's harder to debug"
```suggestion
				errs.AddForProperty(fmt.Sprintf(prefix+".rank_table.%i.start_rank",i), fmt.Errorf("must be greater than start_rank of element #%d", i-1))
```